### DISCUSSION TITLE
Errors Regarding pip install --editable

### DISCUSSION THREAD

**Discussion by zsalimian1359** [iiasa/message_ix#423] [Getting started] [answered]
pip install --editable c:\\users\documents\github\main\message-ix.[docs,reporting,tests,tutorial] shows error.What shall I do?

**Comment by khaeru** [comment]
Please supply the full error message. Without that, it is impossible to answer.

**Comment by vahabmokari** [comment]
Dear Sir,
Although I know that I shouldn't upload any screenshot, but I hadn't any better way to show you my problem.
 I should mention that I know that I can work with Jupyter, I tried Anaconda and Jupyter **_on my friend's laptop_** whom was a beginner user and it worked properly. So, I don't have any problem with installation from Anaconda.
But I would like to work with MESSAGE-ix via python not Jupyter, on my own laptop. Firstly, I should mention that I have Windows7 **_on my own laptop_**.I don't want to contribute to Message-ix so I don't need to clone Github.I just used **_Pip_** via Miniconda. I run $pip install message-ix in the command run. Then I asked for message-ix show-versions. It worked properly.
But After that I called Python
Then I tried to write a code calling message-ix and ixmp
As you see in the following screenshot, after I run mp=ixmp.Platform() it shows the following message which confused me alot. 
![Untitled](https://user-images.githubusercontent.com/73784155/99252989-271c5000-2825-11eb-8f7e-5fb41f2c02cf.png)

**Comment by zsalimian** [comment]
Dear @khaeru 
@vahabmokari  is my colleague. Also, the same errors in the screenshot occurred on my own computer while trying the following command in python 
mp=ixmp.Platform()
Regards

**Comment by zsalimian** [comment]
Moreover, as @vahabmokari explained before, he did not want to contribute to message-ix on Github, thus needn't use the --editable command. But whenever I use pip install --editable C:\Github\main\message-ix.[docs,tests,reporting,tutorial] it shows me the following error
I and @vahabmokari know that we shouldn't use screenshots. But it is the last time. We had many problems regarding installation
![image](https://user-images.githubusercontent.com/71164512/99360521-f5a79100-28c5-11eb-8551-e7bc0cc6fbff.png)

**Comment by zsalimian** [comment]
@khaeru 
Dear Sir,
Is it possible to answer this question?
Regards

**Comment by khaeru** [comment, marked as answer]
This latest screenshot includes an invalid command-line call to `pip`. To repeat https://github.com/iiasa/message_ix/issues/419#issuecomment-720348157, you must understand the pre-requisite knowledge and skills for using MESSAGEix. This includes being familiar with tools like `pip` for installing and uninstalling Python code.

To be very clear, when we say this knowledge is "pre-requisite", it means that the items listed—all of them—are **required** in order to successfully use the software. “Required” means “not optional.”

Regrettably, we do not currently have any resources to provide free-of-charge training or support to people who do not have the pre-requisite knowledge and skills. However, it is not hard to learn them yourself, using the links provided from the documentation. You could also find other online courses or tutorials, or someone at your institution or a contractor who knows them and can teach you.
